### ISSUE TITLE
FOR OUTREACHY UI UPLIFT PROJECT - Identify accessibility improvements for mobile apps

### ISSUE THREAD

**Issue by maprehensive** [opendatakit/tool-suite-X#348]
The scope of this issue is to identify accessibility improvements for Services, Survey or Tables. 

The output of this issue should be a UI design change that improves accessibility in one of the mobile applications (Notify is out of scope for this issue). 

There are two parts to this issue:

1. Identify accessibility improvements using Accessibility Scanner app

- Use the Android [Accessibility Scanner app](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor) app to identify accessibility issues
- Follow the "Learn more: _issue name_" link in Accessibility Scanner to learn how to implement an improvement

2. Design a fix for the issue using Android Studio

- Checkout the project for the mobile app
- Find the UI element and apply the design improvement to fix the accessibility issue
- Build the project, screenshot the improvement in the emulator
- Write a brief description of the accessibility issue and the design that you applied

Then post your solutions back to this issue. 

This issue can be used for an Outreachy application for the project [User interface uplift for mobile apps](https://www.outreachy.org/outreachy-may-2022-internship-round/communities/odk-x/#user-interface-uplift-for-mobile-apps)

Please request to be assigned to the issue so we can track the number of peopling doing this exercise. Multiple people can work on this issue simultaneously.

**Comment by khushi200701** [comment]
Hello, I'm interested in working on this. Please assign me this issue :)

**Comment by wbrunette** [comment]
@khushi200701 I have added you to the issue

**Comment by khushi200701** [comment]
Hey, the link for the Accessibility Scanner app is broken, can't access it :/
![image](https://user-images.githubusercontent.com/58480229/161538851-48e32e3c-b485-432c-8292-2a865da343ee.png)

**Comment by shreyavijayvargiya** [comment]
can i also work on this? will you assign it to me?

**Comment by wbrunette** [comment]
@shreyavijayvargiya  I have added you

**Comment by maprehensive** [comment]
> Hey, the link for the Accessibility Scanner app is broken, can't access it :/
> ![image](https://user-images.githubusercontent.com/58480229/161538851-48e32e3c-b485-432c-8292-2a865da343ee.png)

@khushi200701 try starting with this link: https://support.google.com/accessibility/android/answer/6376570?hl=en
Perhaps it's a Play Store region issue.

**Comment by shreyavijayvargiya** [comment]
@maprehensive @wbrunette 
I made some improvements in the login_fragment of the services app. Please check it and give some feedback if any.

![IMG_20220411_181423](https://user-images.githubusercontent.com/57625355/162746315-b78f2d38-aac0-482a-9db0-1521d34d278f.jpg)

This is the login_fragment of the services app. The accessibility scanner app gave me a few suggestions for this screen- 
1. It suggested increasing the size of the two EditText fields to 48dp.
2. It also suggested adding item labels (or android:hint)  in the EditText fields.

I've also changed the fonts, the background color and the button colors according to the style guide provided. 
I've added the font Montserrat and also added a drawable resource file and defined the gradient colors and shape of the button in it. 
I've also added adequate padding and margins in between elements to give it a more seamless look.
Here is the final design built in android studio emulator-

<img width="182" alt="login" src="https://user-images.githubusercontent.com/57625355/162747918-994f1608-ea1a-4ebe-b2d9-6b33b3247f9f.png">

**Comment by sanskruti-shahu** [comment]
Hey @wbrunette , I am interested please assign me this issue under Outreachy.

**Comment by wbrunette** [comment]
@sanskruti-shahu I have added you

**Comment by maprehensive** [comment]
@shreyavijayvargiya Thanks for the contribution - looking good!

Here are some suggestions for going deeper on this:

- follow up on the Item label issue with the form fields
- Consider whether the blended background creates low visual contrast in parts of the button
- Is there an improvement that could be added to the style guide as a result of your changes?

Thanks!
Ben

**Comment by nuilewis** [comment]
Hello, please I would like to work on this issue for Outreachy

**Comment by wbrunette** [comment]
@nuilewis I have added you

**Comment by nuilewis** [comment]
> @nuilewis I have added you

okay, thanks.

**Comment by aasthaa-007** [comment]
Hey @wbrunette can I work on this issue?

**Comment by niyolynate** [comment]
Am interested in working on this issue. please assign me under Outreachy.

**Comment by wbrunette** [comment]
@aasthaa-007 and @niyolynate I have added you to the issue

**Comment by nuilewis** [comment]
Hello,
I started working on the design for the Services app. the main feedback I got from the Accessibility Scanner was a lack of contrast between the different elements.
So I tried to address that with the design. I cleaned up some of the screens and tried to improve the contrast of the different elements.
I also tried to make some of the pages a bit clearer, and more user friendly, especially for new users.

I have added an image with my updates and thinking, and would like some feedback, please.

![ODK-X Services redesign](https://user-images.githubusercontent.com/72214279/163506557-9267a8ef-35e1-4871-8f56-179335a8fda8.png)


Thanks!

**Comment by niyolynate** [comment]
Have tried with Eclipse. after multiple times failing with Android. #first-time issues. but continuing to try. 
![Welcome](https://user-images.githubusercontent.com/98440370/163898564-f77e6734-ffc2-4323-9bba-0b904ef1b342.png)
 Still on rough view while taking time to try again with android. Network related issues.
![w2](https://user-images.githubusercontent.com/98440370/163899416-5c2d46f8-80f5-4fbf-8e6e-6166746baa8d.png)

**Comment by maprehensive** [comment]
@nuilewis This is very clearly laid out, with some good ideas for improving the legibility of the interface. Nice work. 

For the colors, did you consider the colors in the Design Guidelines?
https://docs.google.com/presentation/d/1QfeE20gnWU9HFWkgSo5aaFMsTItNtQbRqXPu0ZqmeH0/

It is acceptable to suggest improvements to the colors in the Guidelines, but you should have a rationale, and note that the guidelines would need updating if the changes were accepted. 

This is important for the UI Uplift project, as the outcomes are not only about immediate design improvements, but building the guidelines, design patterns and templates for other members of the community to consistently and easily apply design standards. 

Likewise with the font you have chosen. 

So I would include details on how you could integrate the changes with the existing Design Guidelines.  And for your application remember that building designs and templates that others can build on is one of the goals.

**Comment by maprehensive** [comment]
@niyolynate Is the issue that you're unable to download Android Studio due to network problems?

**Comment by nuilewis** [comment]
> @nuilewis This is very clearly laid out, with some good ideas for improving the legibility of the interface. Nice work. 
> 
> For the colors, did you consider the colors in the Design Guidelines?
> https://docs.google.com/presentation/d/1QfeE20gnWU9HFWkgSo5aaFMsTItNtQbRqXPu0ZqmeH0/
> 
> It is acceptable to suggest improvements to the colors in the Guidelines, but you should have a rationale, and note that the guidelines would need updating if the changes were accepted. 
> 
> This is important for the UI Uplift project, as the outcomes are not only about immediate design improvements, but building the guidelines, design patterns and templates for other members of the community to consistently and easily apply design standards. 
> 
> Likewise with the font you have chosen. 
> 
> So I would include details on how you could integrate the changes with the existing Design Guidelines.  And for your application remember that building designs and templates that others can build on is one of the goals. 
> 

Thank you for the feedback. When designing I did not have access to the Style Guide, I was unable to find the link so I could not get the proper colours.

**Comment by shreyavijayvargiya** [comment]
> @shreyavijayvargiya Thanks for the contribution - looking good!
> 
> Here are some suggestions for going deeper on this:
> 
> * follow up on the Item label issue with the form fields
> * Consider whether the blended background creates low visual contrast in parts of the button
> * Is there an improvement that could be added to the style guide as a result of your changes?
> 
> Thanks! Ben

Hey @maprehensive, Thanks for your feedback. I worked on it and made a few changes to the UI-
1. I removed the TextFields and used only the android:hint in EditTextFields as it solves the purpose and saves space.
2.  You were right, the color that I chose earlier wasn't fitting so I took a color that's two shades lighter than the one provided in the style guide from the material design color palette and used it for the gradient. I also changed the gradient angle and now it gives a nice little 3-D effect.
3.  The Improvement that I would suggest currently is to add templates for particular screens instead of just the color schemes and fonts which is also the scope for this project and I plan to cover it in my internship period.
4.  I also wanted to know your opinion if all the headings should be in bold (as shown in the below screenshots) or not (refer sync_launch layout- the third screenshot below).
Attaching the final layouts built in Android emulator-

<img width="184" alt="login_fragment" src="https://user-images.githubusercontent.com/57625355/164125026-cfe50aaa-e89d-4e38-83a0-6b38b31758b5.png">

<img width="184" alt="verify_permiss" src="https://user-images.githubusercontent.com/57625355/164125189-eefcf596-b441-4493-9b0b-60f603b820da.png">

<img width="182" alt="sync_launch" src="https://user-images.githubusercontent.com/57625355/164125209-0a58a0e7-2cc5-4752-b1ea-815963259d78.png">

<img width="179" alt="server_pass" src="https://user-images.githubusercontent.com/57625355/164125221-411e5147-65d4-4980-b240-04d85a15d683.png">

<img width="182" alt="admin_pass" src="https://user-images.githubusercontent.com/57625355/164125243-7f00ae85-eddc-4f08-ab6e-b6262b082b78.png">

Please review these changes and give feedback if any :)
Thanks!

**Comment by khushi200701** [comment]
Hey, I would like to make a small suggestion here, hope you don't mind.

For the first screen, where you removed text labels and added placeholder
text instead, it's not a good UX practice, that's because when someone
starts typing in the text box, the placeholder text disappears and if the
user gets distracted for a second, they wouldn't remember what they were
typing.
I'm sure you'll find some articles on this topic explaining it in detail if
you're interested.

Also, UI is a part of UX, we shouldn't compromise on the UX just to make it
look prettier :)

**Comment by maprehensive** [comment]
@shreyavijayvargiya 
Thanks for iterating on your contribution, some good improvements there. 

For the headings, I think that having contrast between the text field and the labels, or headings and text, is very important for legibility. You're achieving this by bolding the headings.

**Comment by khushi200701** [comment]
@maprehensive I have not posted my design improvements here, instead just directly put them in my proposal.
Should i put them here as well?

**Comment by maprehensive** [comment]
@khushi200701 Yes please! then you can reference this issue in your application. It's part of the Outreachy requirements.

**Comment by khushi200701** [comment]
Hi,
Here are the changes I made for **SURVEY:**
<br/>

![image](https://user-images.githubusercontent.com/58480229/164413107-23aa05f8-b75d-4aff-b262-82e4f86da6ff.png)

Changes made:

- Increased whitespace to give the design some breathing room
  - Left and right padding
  - Padding above and below the heading
- Gave a light blue background color to the heading to distinguish it from the rest of the data.
- Made the overall screen background white instead of grey (white gives a brighter and cleaner look according to me).

**Comment by khushi200701** [comment]
Here are the changes I made for **SERVICES:**        
                                 
![image](https://user-images.githubusercontent.com/58480229/164414651-4b100588-b90a-46ce-a93d-f27a2185370d.png)

Changes made:

- Increased whitespace to give the design some breathing room
  - Left and right padding
  - Padding above and below the headings and text
- Made the text a darker colour to fix the text contrast issue (but not a jet black as it can be a little harsh on the eyes sometimes)
- Made the spacing for the username and password text fields equal for a more uniform look.
- Put the 3 buttons on the bottom of the screen for easier and quicker access.
- Gave the buttons different colours to distinguish and help them stand out as a call to action (CTA).
- Increased button height to improve touch target size.

<br/>

![image](https://user-images.githubusercontent.com/58480229/164415417-53495b34-39fc-4d8f-bc89-4b6494121ee9.png)

Changes made:

- Increased whitespace to give the design some breathing room
  - Left and right padding
  - Padding above and below the headings and text
- Made the text a darker colour to fix the text contrast issue (but not a jet black as it can be a little harsh on the eyes sometimes)
- Put the button on the bottom of the screen for easier and quicker access.
- Gave the button a blue colour to help it stand out as a call to action (CTA)

**Comment by khushi200701** [comment]
![image](https://user-images.githubusercontent.com/58480229/164467519-dff67372-fa6a-4292-93cf-f7642c44b083.png)

For the **Tables** app, I figured out some issues which I wasn't able to solve through Android studio because the Tables app gets the data from the organization and not ODK.

So, for this, we can focus on making basic changes like improving text contrast, white space, visual hierarchy, and text alignment, which I've tried to do here on the menu bar and buttons.

**Comment by niyolynate** [comment]
@maprehensive yes please i have failed totally tried all but nothing. so i tried with eclipse, after multiple times of failure to an issue i had been assigned to.
if anyone could help will appreciate it.
i live in a remote area to network coverage
 and i think also my coding skills are not so good. 
![ODK - X](https://user-images.githubusercontent.com/98440370/164694538-ca7522ea-6c3a-47dc-8e5b-34ca7f565195.png)

**Comment by niyolynate** [comment]
> @niyolynate Is the issue that you're unable to download Android Studio due to network problems?

Yes. i have downloaded multiple times, and the last time it brought me C++,  Kotlin Instead of Java.

**Comment by aasthaa-007** [comment]
Here is one of the issues which I got from the Accessibility Scanner App
In this issue, I fixed the size of the field to make them more readable to the user

![screenshot_ODK-X Services_2022-04-14-16_04_03](https://user-images.githubusercontent.com/70707726/164721533-9463b15f-8619-460a-9007-557f4d57a9e3.png)

Below are the changes made to resolve the issue,

![image](https://user-images.githubusercontent.com/70707726/164727150-d611b863-0549-4165-a5a7-097306115d8a.png)

Also, I added a new field for confirming passwords to meet the standard UI development practices
Please review these changes and provide feedback (if any)

**Comment by maprehensive** [comment]
This issue will be closed as the Outreachy applications have concluded.
